clientconn_authority_test.go

```diff
@@ -47,10 +47,12 @@ func (s) TestClientConnAuthority(t *testing.T) {
 			wantAuthority: "Non-Existent.Server:8080",
 		},
 		{
-			name:          "override-via-creds",
-			target:        "Non-Existent.Server:8080",
-			opts:          []DialOption{WithTransportCredentials(creds)},
-			wantAuthority: serverNameOverride,
+			name:   "override-via-creds",
+			target: "Non-Existent.Server:8080",
+			opts:   []DialOption{WithTransportCredentials(creds)},
+			// Overriding authority from the transport credentials is not
+			// supported anymore.
```

Comment:
Isn't this a behavior change?  "not supported" and "not recommended" and "deprecated" is fine but we shouldn't change behavior that users may be relying upon.  If we're reasonably sure nobody is using this (IIRC we added it only for grpclb?) then maybe we can try it and be ready to revert it, but if it's easy enough to keep the behavior, then we should.

Reply:
That logic seems fine, but then the comment should be a little different here.  It should say the channel authority is not impacted, but the credentials implementation may still validate using that authority instead of the one provided by the channel.

However, is this okay for security reasons, to have the channel thinking the authority is one thing, but the credentials silently using a different authority?  It's a bit of a weird thing in the first place, but it seems like it should notify the channel if it's happening.  Should we at least deprecate all things related to the credentials override?

Reply:
I think we should keep supporting this, unless it's too messy. But it sounds not too bad to me.